backend/layers/processing/h5ad_data_file.py

```diff
@@ -95,19 +98,23 @@ def to_cxg(
 
     def write_anndata_x_matrices_to_cxg(self, output_cxg_directory, ctx, sparse_threshold):
         matrix_container = f"{output_cxg_directory}/X"
-
         x_matrix_data = self.anndata.X
-        is_sparse = is_matrix_sparse(x_matrix_data, sparse_threshold)  # big memory usage
-        logging.info(f"is_sparse: {is_sparse}")
-
-        convert_matrices_to_cxg_arrays(matrix_container, x_matrix_data, is_sparse, ctx)  # big memory usage
+        with dask.config.set(
+            {
+                "num_workers": 2,  # match the number of workers to the number of vCPUs
+                "threads_per_worker": 1,
+                "distributed.worker.memory.limit": "6GB",
+                "scheduler": "threads",
```

Comment:
Change to "processes" to make use of multiple cores when reading the chunked h5ad.

Reply:
based on this dask documentation, I think threads scheduler makes more sense for our use-case (dealing w/ numpy/pandas, tiledb objects) https://docs.dask.org/en/stable/best-practices.html#processes-and-threads

what do you think?

Reply:
I would run a bench mark to see. The chunked reads from the anndata could thread bottle neck.